crates/sui-core/src/consensus_adapter.rs

```diff
@@ -361,23 +363,35 @@ impl ConsensusAdapter {
         }
 
         // Use the minimum digest to compute submit delay.
-        let min_digest = transactions
+        let min_digest_and_gas_price = transactions
             .iter()
             .filter_map(|tx| match &tx.kind {
                 ConsensusTransactionKind::CertifiedTransaction(certificate) => {
-                    Some(certificate.digest())
-                }
-                ConsensusTransactionKind::UserTransaction(transaction) => {
-                    Some(transaction.digest())
+                    Some((certificate.digest(), certificate.gas_price()))
                 }
+                ConsensusTransactionKind::UserTransaction(transaction) => Some((
+                    transaction.digest(),
+                    transaction.data().transaction_data().gas_price(),
+                )),
                 _ => None,
             })
             .min();
 
-        let (duration, position, positions_moved, preceding_disconnected) = match min_digest {
-            Some(digest) => self.await_submit_delay_user_transaction(committee, digest),
-            _ => (Duration::ZERO, 0, 0, 0),
-        };
+        let (duration, position, positions_moved, preceding_disconnected) =
+            match min_digest_and_gas_price {
+                Some((digest, gas_price)) => {
+                    // TODO: Make this configurable.
+                    let k = 5;
+                    let multipler = gas_price / epoch_store.reference_gas_price();
+                    let amplification_factor = if multipler >= k { multipler } else { 0 };
```

Comment:
I think in the original proposal this was `if multipler >= k { multipler - 1 } else { 0 }`, but I like the simpler version here.